### ISSUE TITLE
Are we sure slaves are actually having an impact?

### ISSUE THREAD

**Issue by thelulzy** [ensingm2/SteamMonsterGameScript#109]
I believe the original claim was that someone noticed increased gold years when running multiple instances of the script. This was used as evidence to the potential effectiveness of slaves.

I currently have 2 games running on different computers in the same room, running the newest version of this script. One of these scripts I have running 5 slaves, the other 0 slaves.

I can assure you that the gold inflow rate is equivalent. What evidence do we have that slaves are currently doing anything?

**Comment by Evanito** [comment]
Yeah I never used them because It sounded preposterous.

Just for clarification, your two computers are on different steam accounts?

**Comment by thelulzy** [comment]
Yes, they are

**Comment by ensingm2** [comment]
Yeah, I haven't been 100% on slaves. I know they worked at least for a while on Raining Gold values. Not sure if they still do or not. it's something I need to look into, but isn't top priority at the moment.

At some point I'm planning to just do away with slaves altogether and just send the data to the server multiple times instead, that's cleaner and less intensive.

**Comment by cdmichaelb** [comment]
They seemed to be working with manually created slave (Open New Window on Summer Game Info > Resume Game), but using the slave feature in the script seems to do nothing except lag everything.

It's really hard to tell because of variable use of raining gold. I think it might be best to test it in a slow room. I'm in the billions as far as gold, and I don't see a difference between 10 manual clients and 1.

**Comment by ensingm2** [comment]
https://www.reddit.com/r/SteamMonsterGame/comments/3a0hbc/results_do_slave_windows_work/

So.... Probably not.

They may have implemented a security measure or something to blacklist us. Two possibilities I see in this case:
- There's a validity check tied to the renderer that we disable. This should be easy enough to work around if it's the case.
- They look at the url just like we do in order to identify a slave (`if(location.search.match(/slave/))`)
  - If they were dumb and check for that specific string, we can replace it and move on.
  - If they check for ANY extra string and flag us, we may have to think of a plan B.

**Comment by ensingm2** [comment]
Hmmm... After thinking about it, could this have been the cause?

https://github.com/ensingm2/SteamMonsterGameScript/issues/120

If so, and assuming that linked reddit post works, slaves should be working now?

I'll test tomorrow.

**Comment by cdmichaelb** [comment]
I generated 2 slaves, see if I can notice something, but I think it's too late in this game to easily notice since I'm earning millions with trillions in the wallet. An idea though, you could add some form of debug that estimates how much gold you should have got from raining gold? Then compare it how much you actually got. Then if the number adds up to main client + slaves, it's probably working.... if it adds up to main client + some change, then it's probably not working.

Or simply create a gold per minute(hour?) counter? Then open extra slaves and see if that value jumps as it should.

I just generated 10 slaves, they appear to be functioning, I was gaining 1b per increment so now I should be gaining ~9-10b per increment right? Or at the very least it SHOULD be noticeable?

Here are my money updates in order: (Posted on reddit to make this post not so gigantic and so the other person testing it could see my results.)

https://www.reddit.com/r/SteamMonsterGame/comments/3a0hbc/results_do_slave_windows_work/cs87ypx

**Comment by cdmichaelb** [comment]
Start of today, tried multiple clients, it acts like if you send 400 clicks per minute in a single client...  I'll get a bunch of gold and then it's all revoked. I don't think multi-client works anymore.

**Comment by ensingm2** [comment]
Damn, alright.

I'm gonna do a bit more testing later on and if I get the same results I'll disable slaves for clarity.
